plugins/modules/source_control/gitlab/gitlab_group_variable.py

```diff
@@ -0,0 +1,297 @@
+      description: A list of variables which were deleted.
+      returned: always
+      type: list
+      sample: "['ACCESS_KEY_ID', 'SECRET_ACCESS_KEY']"
+    updated:
+      description: A list of variables whose values were changed.
+      returned: always
+      type: list
+      sample: "['ACCESS_KEY_ID', 'SECRET_ACCESS_KEY']"
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.six import string_types
+from ansible.module_utils.six import integer_types
+
+
+GITLAB_IMP_ERR = None
+try:
+    import gitlab
+    HAS_GITLAB_PACKAGE = True
+except Exception:
+    GITLAB_IMP_ERR = traceback.format_exc()
+    HAS_GITLAB_PACKAGE = False
+
+from ansible_collections.community.general.plugins.module_utils.gitlab import gitlabAuthentication
+
+
+class GitlabGroupVariables(object):
+
+    def __init__(self, module, gitlab_instance):
+        self.repo = gitlab_instance
+        self.group = self.get_group(module.params['group'])
+        self._module = module
+
+    def get_group(self, group_name):
+        return self.repo.groups.get(group_name)
+
+    def list_all_group_variables(self):
+        return self.group.variables.list()
+
+    def create_variable(self, key, value, masked, protected, variable_type):
+        if self._module.check_mode:
+            return
+        return self.group.variables.create({"key": key, "value": value,
+                                            "masked": masked, "protected": protected,
+                                            "variable_type": variable_type})
+
+    def update_variable(self, key, var, value, masked, protected, variable_type):
+        if var.value == value and var.protected == protected and var.masked == masked and var.variable_type == variable_type:
+            return False
+
+        if self._module.check_mode:
+            return True
+
+        if var.protected == protected and var.masked == masked and var.variable_type == variable_type:
+            var.value = value
+            var.save()
+            return True
+
+        self.delete_variable(key)
+        self.create_variable(key, value, masked, protected, variable_type)
+        return True
+
+    def delete_variable(self, key):
+        if self._module.check_mode:
+            return
+        return self.group.variables.delete(key)
+
+
+def native_python_main(this_gitlab, purge, var_list, state, module):
+
+    change = False
+    return_value = dict(added=list(), updated=list(), removed=list(), untouched=list())
+
+    gitlab_keys = this_gitlab.list_all_group_variables()
+    existing_variables = [x.get_id() for x in gitlab_keys]
+
+    for key in var_list:
+        if not isinstance(var_list[key], (string_types, integer_types, float, dict)):
+            module.fail_json(msg="Value of %s variable must be of type string, integer or dict, passed %s" % (key, var_list[key].__class__.__name__))
```

Comment:
```suggestion
            module.fail_json(msg="Value of %s variable must be of type string, integer, float or dict, passed %s" % (key, var_list[key].__class__.__name__))
```
Not to forget `float` :)

In case `var_list[key]` is a `dict,` how about checking the types of `value`, `masked`, `protected` and `variable_type` as well (when provided)?